### ISSUE TITLE
Worker Interface

### ISSUE THREAD

**Issue by kevinconway** [kevinconway/scriptpool#3]
A `worker` is a resource that carries out some action, or series of actions, based on messages placed in its `queue`.

The responsibility of a `worker` is:
- Listen for messages on a `queue`
- Perform some action as a result of the message
- Report the results of that action

The proposed interface for the `worker` is:

> Worker [arguments]
> 
> arguments:
> 
>   --identity: A unique value used to identify the worker (required, no default).
>   --poll:           Idle time once a queue is empty (default .1 seconds).
>   --terminator:     Message to stop the worker (default terminate_worker).

Each implementation must expose this public interface. Additional arguments specific to the implementation may be added, but they must not be necessary for starting a functioning pool. All implementations should have reasonable defaults for their arguments such that a call built using the above interface will successfully create a worker (even if not an optimal one).

In other words, the command Worker with and identity should create a functional worker.

The proposed first implementations of this interface are:

EchoWorker

A worker that simply prints out the message it receives.

BashWorker

A worker that simply evaluates each message as bash.

ProxyWorker

A worker that simply calls a function called `receive_message` with the message content.

Worker implementations should make use of the Queue interface.
